neo/SmartContract/NativeContracts/NeoToken.cs

```diff
@@ -0,0 +1,182 @@
+                        Snapshot.Storages.Delete(key_from);
+                    }
+                    else
+                    {
+                        state_from.Balance -= amount;
+                        storage_from = Snapshot.Storages.GetAndChange(key_from);
+                        storage_from.Value = state_from.ToStruct().Serialize();
+                    }
+                    StorageKey key_to = new StorageKey
+                    {
+                        ScriptHash = Blockchain.NeoToken.ScriptHash,
+                        Key = to
+                    };
+                    StorageItem storage_to = Snapshot.Storages.GetAndChange(key_to, () => new StorageItem
+                    {
+                        Value = new NeoToken_AccountState().ToStruct().Serialize()
+                    });
+                    NeoToken_AccountState state_to = NeoToken_AccountState.FromStruct((Struct)storage_to.Value.DeserializeStackItem(engine.MaxArraySize));
+                    state_to.Balance += amount;
+                    storage_to.Value = state_to.ToStruct().Serialize();
+                }
+            }
+            SendNotification(engine, new StackItem[] { "Transfer", from, to, amount });
+            return true;
+        }
+
+        private bool NeoToken_Initialize()
+        {
+            StorageKey key = new StorageKey
+            {
+                ScriptHash = Blockchain.NeoToken.ScriptHash,
+                Key = Encoding.ASCII.GetBytes("initialized")
+            };
+            if (Snapshot.Storages.TryGet(key) != null) return false;
+            Snapshot.Storages.Add(key, new StorageItem
+            {
+                Value = new byte[] { 1 },
+                IsConstant = true
+            });
+            UInt160 account = Contract.CreateMultiSigRedeemScript(Blockchain.StandbyValidators.Length / 2 + 1, Blockchain.StandbyValidators).ToScriptHash();
+            key = new StorageKey
+            {
+                ScriptHash = Blockchain.NeoToken.ScriptHash,
+                Key = account.ToArray()
+            };
+            Snapshot.Storages.Add(key, new StorageItem
+            {
+                Value = new NeoToken_AccountState { Balance = NeoToken_TotalAmount }.ToStruct().Serialize()
+            });
+            return true;
+        }
+
+        private class NeoToken_AccountState
+        {
+            public BigInteger Balance;
+            public uint BalanceHeight;
+            public BigInteger GasAvailable;
+            public ECPoint[] Votes = new ECPoint[0];
+
+            public static NeoToken_AccountState FromStruct(Struct @struct)
+            {
+                return new NeoToken_AccountState
+                {
+                    Balance = @struct[0].GetBigInteger(),
+                    BalanceHeight = (uint)@struct[1].GetBigInteger(),
+                    GasAvailable = @struct[2].GetBigInteger(),
+                    Votes = @struct[3].GetByteArray().AsSerializableArray<ECPoint>()
```

Comment:
```suggestion
                    Votes = @struct[3].GetByteArray().AsSerializableArray<ECPoint>(Blockchain.StandbyValidators.Length)
```

or

`engine.MaxArraySize`